source/common/router/config_impl.h

```diff
@@ -1079,12 +1105,14 @@ class RouteEntryImplBase : public RouteEntryAndRoute,
   ClusterSpecifierPluginSharedPtr cluster_specifier_plugin_;
   const Http::Code cluster_not_found_response_code_;
   const std::chrono::milliseconds timeout_;
-  const absl::optional<std::chrono::milliseconds> idle_timeout_;
-  const absl::optional<std::chrono::milliseconds> max_stream_duration_;
-  const absl::optional<std::chrono::milliseconds> grpc_timeout_header_max_;
-  const absl::optional<std::chrono::milliseconds> grpc_timeout_header_offset_;
-  const absl::optional<std::chrono::milliseconds> max_grpc_timeout_;
-  const absl::optional<std::chrono::milliseconds> grpc_timeout_offset_;
+  // Durations below are optional in the config. `UnsetDuration` sentinel is used to represent a
+  // value that was not provided.
```

Comment:
add `This takes less memory than using std::optional.`

Reply:
Done